### PR TITLE
Add LongArray

### PR DESCRIPTION
**Pull request by cjkent**
Adds a `LongArray` class that is the equivalent of `IntArray` and `DoubleArray` for `long` values.

### CODE REVIEW

File: modules/collect/src/main/java/com/opengamma/strata/collect/array/LongArray.java [hunk 1]
```diff
@@ -0,0 +1,1297 @@
...
+
+  /**
+   * Returns an instance with each value divided by the specified divisor.
+   * <p>
+   * This is used to divide the contents of this array, returning a new array.
+   * <p>
+   * This is a special case of {@link #map(LongUnaryOperator)}.
+   * This instance is immutable and unaffected by this method.
+   *
+   * @param divisor  the value by which the array applyAsLong() divided
```
Comment (jodastephen) [inline modules/collect/src/main/java/com/opengamma/strata/collect/array/LongArray.java]: Should be "array values are divided"

File: modules/collect/src/main/java/com/opengamma/strata/collect/array/LongArray.java [hunk 2]
```diff
@@ -0,0 +1,1297 @@
...
+   * <pre>
+   *   result = base.mapWithIndex((index, value) -&gt; index * value);
+   * </pre>
+   * <p>
+   * This instance is immutable and unaffected by this method.
+   *
+   * @param function  the function to be applied
+   * @return a copy of this array with the operator applied to the original values
+   */
+  public LongArray mapWithIndex(LongBinaryOperator function) {
```
Comment (jodastephen) [inline modules/collect/src/main/java/com/opengamma/strata/collect/array/LongArray.java]: Wrong type. Should be `IntLongToLongFunction` or similar

File: modules/collect/src/main/java/com/opengamma/strata/collect/function/IntLongConsumer.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ */
+package com.opengamma.strata.collect.function;
+
+/**
+ * An operation consuming two arguments - {@code int} and {@code long}.
+ * <p>
+ * Implementations of this interface will operate using side-effects.
+ */
+@FunctionalInterface
+public interface IntLongConsumer {
```
Comment (jodastephen) [inline modules/collect/src/main/java/com/opengamma/strata/collect/function/IntLongConsumer.java]: Worth checking if these functional interfaces are the only ones you are going to need for `long`.

File: modules/collect/src/main/java/com/opengamma/strata/collect/array/LongArray.java (hunk 1 above)
Comment (cjkent) [reply modules/collect/src/main/java/com/opengamma/strata/collect/array/LongArray.java]: Oops. I hand a find-and-replace operation that went rogue. I thought I'd fixed everything but missed this one.

**Pull request merged**